### ISSUE TITLE
addurls: add --ifsubdataset-exists="warn" (or "error", or "")

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#4968]
ATM `addurls` would warn if a subdataset already exists.  In the case of using `addurls` to update an existing (super)dataset with hundreds/thousands of subdatasets such warning is not desired since floods the screen with not not pertinent warnings (I know that subdatasets already exist). 
`addurls` already has `-ifexists {overwrite|skip}` for individual files, so I named suggested option to be inline with that. 
Note though that we do have `--existing` in `create_sibling*` and `--missing` in `publish`. IMHO at some point we should harmonize how we name such options (with `if` or not, `exists` vs `existing` etc)

**Comment by kyleam** [comment]
> In the case of using `addurls` to update an existing (super)dataset with hundreds/thousands of subdatasets such warning is not desired since floods the screen with not not pertinent warnings (I know that subdatasets already exist).

Yeah, I think this is one of probably at least a few places where it's clear that addurls wasn't designed with updates in mind.  Avoiding the situation you describe is worth another option in my view.

However, given that it seems using addurls for updates will be a common scneario and I don't think there's a risk of data loss (I'd need to check the handling of unlocked files), perhaps it was a mistake to warn here in the first place, and we should demote the log message to the debug level (or, probably better, yield a "notneeded" result).

What do you think?

**Comment by yarikoptic** [comment]
I would have both logged (DEBUG) and yielded a record - they serve different purposes and might even go to different medium (I started to tend to direct logs into a file, especially for <= DEBUG) for future inspection.  Messages live their own live, can be swallowed, suppressed, combined ("X similar...").

**Comment by kyleam** [comment]
> I would have both logged (DEBUG) and yielded a record - they serve different purposes

With the default `datalad.log.result-level` value, the message of notneeded results is logged at the debug level.  But I agree with your general point that they serve different purposes (e.g., gh-3953).

But, setting aside the specifics of how we drop the warning, you're okay with the idea of getting rid of the warning instead of adding an option like `--ifsubdataset-exists`?
